src/firefly/java/edu/caltech/ipac/astro/ibe/IBE.java

```diff
@@ -151,43 +138,41 @@ public File createDataFilePath(IbeDataParam param) throws IOException {
 
     public URL createDataUrl(IbeDataParam param) throws IOException {
 
-        String fpath = param.getFilePath();
+        String url = ibeDataSource.getDataUrl(param);
 
-        String url = ibeDataSource.getIbeHost() + "/data/" +
-                ibeDataSource.getMission() + "/" + ibeDataSource.getDataset() +
-                "/" + ibeDataSource.getTableName() + "/" + fpath;
         String qstr = "";
         if (param.isDoCutout()) {
-            qstr= addUrlParam(qstr, CUT_CENTER, param.getCenter());
-            qstr= addUrlParam(qstr, CUT_SIZE, param.getSize());
-            qstr= addUrlParam(qstr, "gzip", param.isDoZip());
+            qstr = addUrlParam(qstr, CUT_CENTER, param.getCenter());
+            qstr = addUrlParam(qstr, CUT_SIZE, param.getSize());
+            qstr = addUrlParam(qstr, "gzip", param.isDoZip());
         }
 
-        url = url + (qstr.length() > 0 ? "?" + qstr : "");
+        url += (qstr.length() > 0 ? "?" + qstr : "");
```

Comment:
I understand what you are doing now.  Because IBE2 API changes, and some dataset uses v1 while other use v2, you moved the logic to the IbeDataSource.  I can't say I agree with this.  As is, you have IBE's API logic in 3 places.  v1 is in BaseIbeDataSource,  v2 is in AtlasDataSource, and what's common between them is in IBE.  You should have made IBE handle v1 as well as v2 and have the IbeDataSource identify which version they use.

Reply:
Yes. There should be a second phase where we deprecate IBE1 and use only IBE2 (what i call for now Atlas) but it was not plan to do it now because it has much more impacts and need attention from everybody and changes in the server and client. I postponed deliberately this phase for next year, specially also because IBE2 API is still not fully working and metadata search is changing as we speak. differences from v1 to v2 is not only the api url, it is also the way we getting the files from the result, it's different from datasets: WISE, PTF and 2MASS is one group, ATLAS is another group. I'll raise ticket to work on the refactor and move away from IBE1 when we get time. We should also be implementing a TAP processor, SIA2 and SSA processors.